### PR TITLE
CXP-739 Create a new makefile target to create a coverage summary

### PR DESCRIPTION
**Pull request by romaindak**
**Description (for Contributor and Core Developer)**

In the octolabs https://github.com/akeneo/pim-community-dev/pull/14520
We were able to produce coverage reports for each type of tests in Connectivity.
We want to be able to see the total of merged coverage, over Connectivity Context.
Once it’s done, it will be easy to find which part of the Connectivity codebase are not covered and fix it.

**Definition Of Done (for Core Developer only)**

We have a makefile target `connectivity-connection-coverage` which create a folder with a html render of the total coverage (merging unit, integration & E2E tests).
The target is not executed by the CI, only manually on demand.
The existing targets are not affected.

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

**Review by tomglvng — APPROVED** [review]

**Comment by tseho** [comment]
wait before merging, I need to check something about the deps

File: src/Akeneo/Connectivity/Connection/back/tests/Acceptance/behat-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    extensions:
+        Pim\Behat\Extension\PimFormatter\PimFormatterExtension: ~
+        Behat\Symfony2Extension:
+            kernel:
+                bootstrap: 'config/bootstrap.php'
+                class: 'Kernel'
+                path: 'src/Kernel.php'
+                env: test_fake
+        DVDoug\Behat\CodeCoverage\Extension:
+            cache: coverage/Connectivity/Back/Acceptance/.cache/
```
Comment (pchasle) [inline src/Akeneo/Connectivity/Connection/back/tests/Acceptance/behat-coverage.yml:11]: Will the `.cache` folder be ignored by git once created?

**Review by pchasle — APPROVED** [review]

File: src/Akeneo/Connectivity/Connection/back/tests/Acceptance/behat-coverage.yml (hunk 1 above)
Comment (romaindelouard) [reply src/Akeneo/Connectivity/Connection/back/tests/Acceptance/behat-coverage.yml:11]: The folder `coverage` is ignored.
https://github.com/akeneo/pim-community-dev/blob/master/.gitignore#L34

File: make-file/connectivity-connection.mk [hunk 2]
```diff
@@ -85,6 +85,45 @@ else
...
+		--coverage-clover coverage/Connectivity/Back/EndToEnd/coverage.cov \
+		--coverage-php coverage/Connectivity/Back/EndToEnd/coverage.php \
+		--coverage-html coverage/Connectivity/Back/EndToEnd/ \
+		--testsuite EndToEnd $(0)
+	# run the backend application acceptance tests on scope connectivity
+	XDEBUG_MODE=coverage $(PHP_RUN) -d memory_limit=-1 vendor/bin/behat \
+			--config src/Akeneo/Connectivity/Connection/back/tests/Acceptance/behat-coverage.yml \
+			--format pim --out var/tests/behat/connectivity/connection --format progress --out std --colors
+	# download phpcov binary
+	test -e phpcov.phar || wget https://phar.phpunit.de/phpcov.phar
```
[the remark this answers is not in the document]
Comment (romaindelouard) [reply make-file/connectivity-connection.mk:112]: Yes, I can.

File: make-file/connectivity-connection.mk [hunk 3]
```diff
@@ -85,6 +85,46 @@ else
...
+		--testsuite EndToEnd $(0)
+	# run the backend application acceptance tests on scope connectivity
+	XDEBUG_MODE=coverage $(PHP_RUN) -d memory_limit=-1 vendor/bin/behat \
+			--config src/Akeneo/Connectivity/Connection/back/tests/Acceptance/behat-coverage.yml \
+			--format pim --out var/tests/behat/connectivity/connection --format progress --out std --colors
+	# download phpcov binary
+	$(DOCKER_COMPOSE) run -u www-data --rm php test -e phpcov.phar || wget https://phar.phpunit.de/phpcov.phar && \
+		php phpcov.phar --version
+	# create a coverage global folder
+	$(DOCKER_COMPOSE) run -u www-data --rm php sh -c "\
```
Comment (tseho) [inline make-file/connectivity-connection.mk:115]: Why are you running this command docker ?

Comment (romaindak) [reply make-file/connectivity-connection.mk:115]: I prefer run this command in the container with docker-compose.
If the volume mapping is not done then the command will always work.

**Review by tseho — APPROVED** [review]

**Comment by tseho** [comment]
You can rebase and merge this one, the new vendors are accepted.

**Pull request merged**